src/executor/validate.rs

```diff
@@ -0,0 +1,204 @@
+        .fetch_schema(table_name)
+        .await?
+        .ok_or(ExecuteError::TableNotExists)?;
+    let column_indexes = match column_validation {
+        ColumnValidation::All => fetch_all_unique_column_indexes(&column_defs),
+        ColumnValidation::SpecifiedColumns(columns) => {
+            fetch_specified_unique_column_indexes(&column_defs, &columns)
+        }
+    };
+    let unique_checks = create_unique_checks(&column_defs, &column_indexes, items, item_to_row)?;
+    if unique_checks.is_empty() {
+        return Ok(());
+    }
+
+    let column_defs = Rc::new(column_defs);
+    let unique_checks = Rc::new(unique_checks);
+    storage
+        .scan_data(table_name)
+        .await
+        .map(stream::iter)?
+        .try_for_each(|(_, row)| {
+            let column_defs = Rc::clone(&column_defs);
+            let unique_checks = Rc::clone(&unique_checks);
+
+            async move {
+                for check in unique_checks.as_ref() {
+                    let column_index = check.column_index;
+                    let row_val = row.get_value(column_index).unwrap();
+                    if check
+                        .values
+                        .binary_search_by(|v| compare_values_for_sort(v, row_val).unwrap())
+                        .is_ok()
+                    {
+                        return Err(ValueError::DuplicateEntryOnUniqueField(
+                            format!("{:?}", row_val),
+                            column_defs[column_index].name.to_string(),
+                        )
+                        .into());
+                    }
+                }
+                Ok(())
+            }
+        })
+        .await
+}
+
+fn fetch_all_unique_column_indexes(column_defs: &[ColumnDef]) -> Vec<usize> {
+    let mut column_indexes = vec![];
+    for (i, col) in column_defs.iter().enumerate() {
+        if col
+            .options
+            .iter()
+            .any(|opt_def| matches!(opt_def.option, ColumnOption::Unique { .. }))
+        {
+            column_indexes.push(i);
+        }
+    }
+
+    column_indexes
+}
+
+fn fetch_specified_unique_column_indexes(
+    table_column_defs: &[ColumnDef],
+    specified_columns: &[Ident],
+) -> Vec<usize> {
+    let mut column_indexes = vec![];
+    for (i, table_col) in table_column_defs.iter().enumerate() {
+        if table_col
+            .options
+            .iter()
+            .any(|opt_def| match opt_def.option {
+                ColumnOption::Unique { .. } => specified_columns
+                    .iter()
+                    .any(|specified_col| specified_col.value == table_col.name.value),
+                _ => false,
+            })
+        {
+            column_indexes.push(i);
+        }
+    }
+
+    column_indexes
+}
+
+#[derive(Debug)]
+struct UniqueCheck<'a> {
+    column_index: usize,
+    values: Vec<&'a Value>,
+}
+
+fn create_unique_checks<'a, U: 'static + Debug, F>(
+    table_column_defs: &[ColumnDef],
```

Comment:
Only subset of `table_column_defs` is used, and it is only limited to `unique_column_indexes`.
So, it would be better to be merged into a single param.
You can use something like... `unique_columns: &[(usize, ColumnDef)]`, then you don't need to get unnecessary extra params into this function.

Then, you can remove an unnecessary `panic!()` possible case in here below,
`163 table_column_defs[column_index].name.to_string(),`
You can simply replace from `table_column_defs[column_index]` to `column_def` which is panic safe.